Add doc for filter support for cli tools (#11012)
gravitational/teleport: docs/pages/access-controls/reference.mdx

Refer to the [Second Factor - U2F guide](./guides/u2f.mdx) if you have a cluster
using the legacy U2F support.

> ## Filter fields explained

Reviewer: ```suggestion
## Filter fields
```

> Explanation of the fields used in this guide for the `where` and `filter` conditions.

Reviewer: ```suggestion
Here is an explanation of the fields used in the `where` and `filter` conditions within this guide.
```

| Field                      | Description                                         |
|----------------------------|-----------------------------------------------------|

> | `user.roles`               | the `list` of roles assigned to a user              |

Reviewer: ```suggestion
| `user.roles`               | The list of roles assigned to a user              |
```
I don't think `list` needs to be in code style here since it refers to a type rather than a specific keyword used in a config resource.

> | `session.participants`     | the `list` of participants from a session recording |

Reviewer: ```suggestion
| `session.participants`     | The list of participants from a session recording |
```

> | `ssh_session.participants` | the `list` of participants from a SSH session       |

Reviewer: ```suggestion
| `ssh_session.participants` | The list of participants from an SSH session       |
```

> | `user.metadata.name`       | the user's name                                     |

Reviewer: ```suggestion
| `user.metadata.name`       | The user's name                                     |
```
